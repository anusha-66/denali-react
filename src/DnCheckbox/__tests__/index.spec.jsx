/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render, mount } from 'enzyme';
import { DnCheckbox } from '../../index'; // src/index.tsx

describe('DnCheckbox Tests', () => {
    it('should render with default props', () => {
        expect.assertions(2);

        const wrapper = mount(<DnCheckbox id="checkbox1" />);

        expect(wrapper.render()).toMatchSnapshot('before change');

        wrapper.find('input[type="checkbox"]').simulate('change');

        expect(wrapper.render()).toMatchSnapshot('after change');
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnCheckbox
                id="checkbox1"
                name="checkbox"
                defaultChecked
                checkboxLabel="Checkbox"
                value="value1"
                isPartial
                isDisabled
                className="additional--class"
            />,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should call on change correctly', () => {
        expect.assertions(1);

        const mockedOnChange = jest.fn();
        const wrapper = mount(<DnCheckbox onChange={mockedOnChange} />);
        wrapper.find('input[type="checkbox"]').simulate('change');

        expect(mockedOnChange).toHaveBeenCalledTimes(1);
    });
});
