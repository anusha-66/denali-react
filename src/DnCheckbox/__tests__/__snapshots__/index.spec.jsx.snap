// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnCheckbox Tests should render correctly with custom props 1`] = `
<div
  class="checkbox additional--class"
>
  <input
    checked=""
    data-partial=""
    disabled=""
    id="checkbox1"
    name="checkbox"
    type="checkbox"
  />
  <label
    for="checkbox1"
  >
    Checkbox
  </label>
</div>
`;

exports[`DnCheckbox Tests should render with default props: after change 1`] = `
<div
  class="checkbox"
>
  <input
    id="checkbox1"
    type="checkbox"
    value=""
  />
  <label
    for="checkbox1"
  />
</div>
`;

exports[`DnCheckbox Tests should render with default props: before change 1`] = `
<div
  class="checkbox"
>
  <input
    id="checkbox1"
    type="checkbox"
    value=""
  />
  <label
    for="checkbox1"
  />
</div>
`;
